src/AdvancedVI.jl

```diff
@@ -47,6 +55,10 @@ function value_and_gradient!(
     return out
 end
```

Comment:
I wonder if we still need this internal function, now that we are fully relying on `DifferentiationInterface`. The only argument for this would be the use of `DiffResults`, but not sure how much this helps.

Reply:
Perhaps it is better to keep this function for internal use -- e.g. it can print an informational message for incorrect `AutoEnzyme` types.